<launch>
  <arg name="robot_name" default="1" />
  <arg name="goalX" default="6" />
  <arg name="goalY" default="1" />
  <arg name="robot_color" default="Red" />


  <!-- <group ns="RigidBody$(arg robot_name)"> -->
  <group ns="nexus$(arg robot_name)_goal">
    <param name="tf_prefix" value="nexus$(arg robot_name)_goal" />
    <!-- <param name="tf_prefix" value="RigidBody$(arg robot_real_id)" /> -->
    <param command="$(find xacro)/xacro $(find marmot)/urdf/goalPole.xacro robot_color:=$(arg robot_color)" name="robot_description"/>

    <node name="roboTFNode" pkg="marmot" type="spoofGoalTransform.py" output="screen" clear_params="true">
      <param name="name" value="$(arg robot_name)" /> 
      <param name="startX" value="$(arg goalX)" /> 
      <param name="startY" value="$(arg goalY)" /> 
    </node>
    
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
    
  </group>
  
  </launch>
